src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -94,7 +98,23 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
         @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
         @Valid @RequestParam(value = "limit", required = false, defaultValue = "1000") Integer limit
     ) {
-        SearchQueryResultModel searchQueryResultModel = new SearchQueryResultModel();
+        List<String> accessibleIds =
+                iamService.listAuthorizedResources(getAuthenticatedInfo(), IamResourceType.DATASNAPSHOT)
+                        .stream()
+                        .map(UUID::toString)
+                        .collect(Collectors.toList());
+
+        List<String> requestIds = searchQueryRequest.getSnapshotIds();
+
+        for (String snapShotId : requestIds) {
+            if (!accessibleIds.contains(snapShotId)){
+                throw new IamForbiddenException("User '" + getAuthenticatedInfo().getEmail()
```

Comment:
I seem to recall seeing other TDR code where inaccessible objects were filtered out but no error was raised. Either was is fine for now though.